packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,185 @@
+        },
+      };
+    }
+
+    // What the physical resource ID is for Cloudformation
+    // Designed so that, if the construct name, percentage, or interval changes,
+    // the deployment config will be deleted and a new one created in its place
+    const resourceName = this.generatePhysicalResourceId(props);
+
+    this.deploymentConfigName = this.generateDeploymentConfigName(props);
+    this.deploymentConfigArn = arnForDeploymentConfig(this.deploymentConfigName);
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, 'DeploymentConfig', {
+      onCreate: { // Run on creation only, to make the resource
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: this.deploymentConfigName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // This `resourceName` is the initial physical ID of the config
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onUpdate: { // Run on stack update
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: this.deploymentConfigName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // If `resourceName` is different from the last stack update (or creation),
+        // the old config gets deleted and the new one is created
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onDelete: { // Run on deletion, or on resource replacement
+        service: 'CodeDeploy',
+        action: 'deleteDeploymentConfig',
+        parameters: {
+          deploymentConfigName: this.deploymentConfigName,
+        },
+      },
+      // Least permissions, only have permission to create or delete this exact config
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+  }
+
+  // Generates the name of the deployment config. It's also what you'll see in the AWS console
+  private generateDeploymentConfigName(props: CustomLambdaDeploymentConfigProps): string {
+    let configName;
+    if (props.deploymentConfigName !== undefined) {
+      configName = props.deploymentConfigName;
+    } else {
+      const id = this.node.uniqueId;
+      const intervalMinutes = props.interval.toMinutes().toString();
+      const percentage = props.percentage.toString();
+      const deploymentType = props.type.toString();
+      if (props.type !== CustomLambdaDeploymentConfigType.LINEAR) {
+        configName = `${id}.Lambda${deploymentType}${percentage}Percent${intervalMinutes}Minutes`;
+      } else {
+        configName = `${id}.Lambda${deploymentType}${percentage}PercentEvery${intervalMinutes}Minutes`;
+      }
+    }
+    return configName;
```

Comment:
I have to say, we have very different code sensibilities 🙂.

Here's how I would write this method:

```suggestion
    return props.deploymentConfigName !== undefined
      ? props.deploymentConfigName
      : `${this.node.uniqueId}.Lambda${props.type}${props.percentage}Percent${props.type === CustomLambdaDeploymentConfigType.LINEAR ? 'Every' : ''}${props.interval.toMinutes()}Minutes`;
```

3 lines instead of 15, no temporary variables.

And now since the method is one expression, I would actually get rid of it completely, as it's only used in a single place, and inline it there 🙂.